include/fiction/utils/math_utils.hpp

```diff
@@ -7,17 +7,19 @@
 
 #include <cmath>
 #include <cstdint>
+#include <cstdlib>
+#include <type_traits>
 
 namespace fiction
 {
 
 /**
  * Rounds a number to a specified number of decimal places.
  *
- * @tparam T the type of the number to round.
- * @param number the number to round.
- * @param n the number of decimal places to round to.
- * @return the number rounded to n decimal places.
+ * @tparam T The type of the number to round.
+ * @param number The number to round.
+ * @param n The number of decimal places to round to.
+ * @return The number rounded to n decimal places.
  */
 template <typename T>
 T round_to_n_decimal_places(const T number, const uint64_t n)
```

Comment:
I know you didn't touch this one, but could you add `noexcept` for consistency?

```suggestion
T round_to_n_decimal_places(const T number, const uint64_t n) noexcept
```